Mathlib/Analysis/Polynomial/CauchyBound.lean

```diff
@@ -0,0 +1,127 @@
+/-
+Copyright (c) 2024 Daniel Weber. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Daniel Weber
+-/
+import Mathlib.Algebra.GeomSum
+import Mathlib.Algebra.Polynomial.Eval
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.Algebra.Polynomial.Monic
+
+/-!
+# Cauchy's bound on polynomial roots.
+-/
+
+variable {K : Type*} [NormedDivisionRing K]
+
+open Polynomial Finset NNReal
+
+/--
+Cauchy's bound on the roots of a given polynomial.
+-/
+noncomputable def cauchyBound (p : K[X]) : ℝ≥0 :=
+  sup (range p.natDegree) (‖p.coeff ·‖₊) / ‖p.leadingCoeff‖₊ + 1
+
+@[simp]
+lemma one_le_cauchyBound (p : K[X]) : 1 ≤ cauchyBound p := by
+  simp [cauchyBound]
+
+@[simp]
+lemma cauchyBound_zero : cauchyBound (0 : K[X]) = 1 := by
+  simp [cauchyBound]
+
+@[simp]
+lemma cauchyBound_C (x : K) : cauchyBound (C x) = 1 := by
+  simp [cauchyBound]
+
+@[simp]
+lemma cauchyBound_one : cauchyBound (1 : K[X]) = 1 := cauchyBound_C 1
+
+@[simp]
+lemma cauchyBound_X : cauchyBound (X : K[X]) = 1 := by
+  simp [cauchyBound]
+
+@[simp]
+lemma cauchyBound_X_add_C (x : K) : cauchyBound (X + C x) = ‖x‖₊ + 1 := by
+  simp [cauchyBound]
+
+@[simp]
+lemma cauchyBound_X_sub_C (x : K) : cauchyBound (X - C x) = ‖x‖₊ + 1 := by
+  simp [cauchyBound]
+
+@[simp]
+lemma cauchyBound_smul {x : K} (hx : x ≠ 0) (p : K[X]) : cauchyBound (x • p) = cauchyBound p := by
+  simp only [cauchyBound, (isRegular_of_ne_zero hx).left.isSMulRegular,
+    natDegree_smul_of_smul_regular, coeff_smul, smul_eq_mul, nnnorm_mul, ← mul_finset_sup,
+    leadingCoeff_smul_of_smul_regular, add_left_inj]
+  apply mul_div_mul_left
+  simpa
+
+lemma NNReal.geom_sum {x : ℝ≥0} (h : 1 < x) (n : ℕ) :
+    ∑ i ∈ Finset.range n, x ^ i = (x ^ n - 1) / (x - 1) := by
+  apply eq_div_of_mul_eq (tsub_pos_of_lt h).ne'
+  apply eq_tsub_of_add_eq
+  convert geom_sum_mul_add (x - 1) n <;>
+  rw [tsub_add_cancel_of_le h.le]
```

Comment:
I think this could move to an earlier file. What does `#find_home!` say?

Reply:
It can't find anything, and adding `import Mathlib` it says `Mathlib.Topology.ContinuousMap.StarOrdered`

Reply:
I created a new file with a slight generalization of this result. I took the liberty of pushing directly to your branch.